frontend-react/src/pages/tos-sign/TermsOfServiceForm.tsx

```diff
@@ -0,0 +1,446 @@
+import { getStates } from "../../utils/OrganizationUtils";
+
+import SuccessPage from "./SuccessPage";
+import { Jwt, JwtPayload, sign, SignOptions } from "jsonwebtoken";
+
+export interface AgreementBody {
+    title: string;
+    firstName: string;
+    lastName: string;
+    email: string;
+    territory: string;
+    organizationName: string;
+    operatesInMultipleStates: boolean;
+    agreedToTermsOfService: boolean;
+}
+
+function TermsOfServiceForm() {
+    const STATES = getStates();
+
+    /* Form field values are stored here */
+    const [title, setTitle] = useState("");
+    const [firstName, setFirstName] = useState("");
+    const [lastName, setLastName] = useState("");
+    const [email, setEmail] = useState("");
+    const [territory, setTerritory] = useState(STATES[0].toLowerCase());
+    const [organizationName, setOrganizationName] = useState("");
+    const [multipleStates, setMultipleStates] = useState(false);
+    const [agree, setAgree] = useState(false);
+    const [success, setSuccess] = useState(false);
+
+    /* The proper flags are set to true if form is submitted without required fields */
+    const [firstNameErrorFlag, setFirstNameErrorFlag] = useState(false);
+    const [lastNameErrorFlag, setLastNameErrorFlag] = useState(false);
+    const [emailErrorFlag, setemailErrorFlag] = useState(false);
+    const [territoryErrorFlag, setterritoryErrorFlag] = useState(false);
+    const [organizationNameErrorFlag, setorganizationNameErrorFlag] =
+        useState(false);
+    const [agreeErrorFlag, setAgreeErrorFlag] = useState(false);
+    const [sendGridErrorFlag, setSendGridErrorFlag] = useState({
+        isError: false,
+        status: 200,
+    });
+
+    const [submitting, setSubmitting] = useState(false);
+
+    const handleSubmit = async (e: any) => {
+        e.preventDefault();
+        setSubmitting(true);
+        resetAllErrorFlags();
+        const body = createBody();
+        if (body === null) {
+            setSubmitting(false);
+            return;
+        }
+        const auth = createAuth()
+        console.log(auth)
+        const response = await fetch(
+            `${AuthResource.getBaseUrl()}/api/email-registered`,
+            {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json",
+                    "Authorization": auth
+                },
+                body: JSON.stringify(body),
+            }
+        );
+        if (response.status < 200 || response.status > 299) {
+            setSubmitting(false);
+            setSendGridErrorFlag({
+                isError: true,
+                status: response.status,
+            });
+            return;
+        }
+        setSuccess(true);
+    };
+
+    const createAuth = (): string => {
+        
+        let payload: object = {
+            sub: window.location.origin
+        }
+        return sign(payload, 'L78wIGpvMQnb7l3N5aplwaptSLHBJ1jb8ytKkNRx5zk=')
```

Comment:
Hey, this is a dummy secret (i.e. not the real secret we're using on the backend) that I plugged in to push this up to version control, this has since been refactored out in favor of using an environment variable